File: 0917-reverse-only-letters/0917-reverse-only-letters.cpp
class Solution {
public:
    string reverseOnlyLetters(string s) {
        vector<char> letters;

        for (char c : s) {
            if (isalpha(c)) {
                letters.push_back(c);
            }
        }

        string result;

        for (char c : s) {
            if (isalpha(c)) {
                result.push_back(letters.back());
                letters.pop_back();
            } else {
                result.push_back(c);
            }
        }
        return result;
    }
};

File: 0917-reverse-only-letters/0917-reverse-only-letters.py
class Solution:
    def reverseOnlyLetters(self, s: str) -> str:
        words = [word for word in s if word.isalpha()]
        words = words[::-1]
        result = []
        for c in s:
            if c.isalpha():
                result.append(words.pop(0))
            else:
                result.append(c)
        return "".join(result)